src/H5Tconv.c

```diff
@@ -2952,6 +2968,10 @@ H5T__conv_enum_init(const H5T_t *src, const H5T_t *dst, H5T_cdata_t *cdata, cons
         if (NULL == (priv->dst_copy = H5T_copy(dst, H5T_COPY_ALL)))
             HGOTO_ERROR(H5E_DATATYPE, H5E_CANTCOPY, FAIL, "unable to copy destination datatype");
 
+        /* Nothing more to do if enum has no members */
+        if (0 == src->shared->u.enumer.nmembs)
```

Comment:
Moved the check for 0 enum members down so that the cached private data still gets a copy of the source and destination datatypes in case `H5T_cmp` is called later.